sdk/resourcemanager/Azure.ResourceManager/src/Internal/MgmtTelemetryPolicy.cs

```diff
@@ -19,6 +18,11 @@ public MgmtTelemetryPolicy(object source, ClientOptions options)
         public override void OnSendingRequest(HttpMessage message)
         {
             var header = message.TryGetProperty("UserAgentOverride", out var userAgent) ? userAgent as string : _defaultHeader;
+            if (message.Request.Headers.TryGetValues(HttpHeader.Names.UserAgent, out var customHeaders))
```

Comment:
A little bit confused, if it's 'override' then why need to append? Or this 'override' only override our default user agent string?

Reply:
I think the logic has changed here. Previously we do override the `user-agent`. But now, we cannot override.

However, `UserAgentOverride` is set in the generated SDK codes, like below:
```csharp
            request.Uri = uri;
            request.Headers.Add("Accept", "application/json");
            message.SetProperty("UserAgentOverride", _userAgent);
```

We can consider to change the name of the property.

Reply:
renamed to `SDKUserAgent`